src/transaction/log_recovery.c

```diff
@@ -754,7 +755,8 @@ log_recovery (THREAD_ENTRY * thread_p, int ismedia_crash, time_t * stopat)
   LSA_COPY (&log_Gl.final_restored_lsa, &log_Gl.hdr.append_lsa);
 #endif /* SERVER_MODE */
 
-  log_append_empty_record (thread_p, LOG_DUMMY_CRASH_RECOVERY, NULL);
+  if ((SERVER_TYPE)prm_get_integer_value(PRM_ID_SERVER_TYPE) != SERVER_TYPE_PAGE)
+    log_append_empty_record (thread_p, LOG_DUMMY_CRASH_RECOVERY, NULL);
```

Comment:
1. `get_server_type ()` can be used to interrogate the server type without reading the system parameter.
2. the preferred coding style is to always add brackets, even for one-line scopes. it makes scopes more visible.
3. make sure automatic indentation works

```suggestion
  if (get_server_type () != SERVER_TYPE_PAGE)
    {
      log_append_empty_record (thread_p, LOG_DUMMY_CRASH_RECOVERY, NULL);
    }
```